core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -79,18 +79,27 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
           return this._subtopics[i];
         }
       }
+      return false;
     };
 
     // The following function is only meant to be called from the undo of
     // a delete subtopic operation and not anywhere else. For adding subtopics
     // in general, use topic.addSubtopic()
     Topic.prototype.undoDeleteSubtopic = function(
-        id, title, skillIdsForSubtopic, isNewlyCreated) {
+        id, title, skillIdsForSubtopic, isNewlyCreated, subtopicIndex) {
       var newSubtopic = SubtopicObjectFactory.createFromTitle(id, title);
       for (var i = 0; i < skillIdsForSubtopic.length; i++) {
         newSubtopic.addSkillId(skillIdsForSubtopic[i]);
       }
+      for (var i = subtopicIndex; i < this._subtopics.length; i++) {
```

Comment:
Shouldn't we be incrementing based on subtopic ID being equal to or larger than the deleted subtopic ID? Not sure what subtopicIndex has to do with it, unless there's some kind of implicit assumption based on the position of the subtopic within the list, which it would be better not to have since you already have subtopic ID as a source of truth.

Make sure to test this stuff really carefully, btw, with different cases.

Reply:
Yeah, I was sorting the array before, but still subtopicId would be better, done.